### PR TITLE
fix 8-bit multi-gpu training bug 

### PR DESCRIPTION
**Pull request by fancyerii**
see https://github.com/huggingface/trl/issues/1348

### CODE REVIEW

File: examples/research_projects/stack_llama_2/scripts/dpo_llama2.py [hunk 1]
```diff
@@ -74,6 +75,11 @@ class ScriptArguments:
         },
     )
 
+    device_map_local_process_idx: Optional[bool] = field(
+        default=True, metadata={"help": "whether to device map model to local process index, see "
+                                "https://github.com/huggingface/trl/issues/1348"}
+    )
```
Comment (younesbelkada) [inline examples/research_projects/stack_llama_2/scripts/dpo_llama2.py:81]:
```suggestion
```
I think we can remove that and always create `device_map = "device_map": {"": Accelerator().local_process_index}`

Can you just make `gradient_checkpointing_kwargs` configurable here? 🙏

**Review by younesbelkada — COMMENTED** [review]
Thanks very much @fancyerii ! I left one comment about making `gradient_checkpointing_kwargs` configurable, I think after that we can ship your PR ! 🚀

File: examples/research_projects/stack_llama_2/scripts/dpo_llama2.py (hunk 1 above)
Comment (fancyerii) [reply examples/research_projects/stack_llama_2/scripts/dpo_llama2.py:85]: updated.

**Comment by younesbelkada** [comment]
Thanks @fancyerii ! 
Could you run the styling checks? `make precommit`

**Comment by fancyerii** [comment]
> Thanks @fancyerii ! Could you run the styling checks? `make precommit`

done. 
btw, when I run `make test`, there are many warnings about "wait for wandb.init"

**Review by younesbelkada — APPROVED** [review]
Thanks again !

**Pull request merged**